libraries/rodeos/include/b1/rodeos/callbacks/coverage.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <b1/rodeos/callbacks/vm_types.hpp>
+#include <eosio/coverage.hpp>
+
+namespace b1::rodeos {
+
+class coverage_maps {
+   public:
+      static coverage_maps& instance() {
+         static coverage_maps instance;
+         return instance;
+      }
+      coverage_maps(const coverage_maps&) = delete;
+      void operator=(const coverage_maps&) = delete;
+
+      cov_map_t funcnt_map;
+      cov_map_t linecnt_map;
+   private:
+      coverage_maps() = default;
+};
+
+struct coverage_state {
+};
+
+template <typename Derived>
+struct coverage_callbacks {
+   Derived& derived() { return static_cast<Derived&>(*this); }
+
+   uint32_t coverage_getinc(uint64_t code, uint32_t file_num, uint32_t func_or_line_num, uint32_t mode, bool inc) {
+      if(inc) {
+         if (mode == 0) {
```

Comment:
An enum class or at least comments would make these `mode` easier to follow.

Reply:
In the host function I'm leaving as uint32_t, but internally will be treated as a enum to improve readability.  It doesn't seem like a great idea to use non-standard types for the host function